### PR TITLE
Propagate resource requirements config to all init containers and update jobs

### PR DESCRIPTION
**Pull request by ramonheras**
fixes: #613 

This MR:
- Propgates `OpenSearchCluster.Spec.InitHelper. Resources` config to all init containers missing resource requirements.
  - [NewSTSForNodePool.keystoreInitContainer](https://github.com/Opster/opensearch-k8s-operator/blob/7bdbfc08fe92193ad792508f27ea5eff1eefdfc4/opensearch-operator/pkg/builders/cluster.go#L322)
  - [NewSTSForNodePool.SetVMMaxMapCount](https://github.com/Opster/opensearch-k8s-operator/blob/7bdbfc08fe92193ad792508f27ea5eff1eefdfc4/opensearch-operator/pkg/builders/cluster.go#L472)
  - [NewBootstrapPod.init](https://github.com/Opster/opensearch-k8s-operator/blob/7bdbfc08fe92193ad792508f27ea5eff1eefdfc4/opensearch-operator/pkg/builders/cluster.go#L751)
  - [NewBootstrapPod.SetVMMaxMapCount](https://github.com/Opster/opensearch-k8s-operator/blob/7bdbfc08fe92193ad792508f27ea5eff1eefdfc4/opensearch-operator/pkg/builders/cluster.go#L810C21-L810C37)
  - [NewSnapshotRepoconfigUpdateJob.snapshotrepoconfig](https://github.com/Opster/opensearch-k8s-operator/blob/7bdbfc08fe92193ad792508f27ea5eff1eefdfc4/opensearch-operator/pkg/builders/cluster.go#L927)
- Creates a new OS Cluster API field `instance.Spec.Security.Config.UpdateJob.Resources` to allow specifying resources for [NewSecurityconfigUpdateJob.updater](https://github.com/Opster/opensearch-k8s-operator/blob/7bdbfc08fe92193ad792508f27ea5eff1eefdfc4/opensearch-operator/pkg/builders/cluster.go#L989).

### CODE REVIEW

**Review by swoehrl-mw — CHANGES_REQUESTED** [review]
Didn't have time to look at it in detail yet. But:

I do not want to have a separate `SnapshotConfig` in the CRD. 
The current mode with a snapshot config job is temporary for a beta feature and will be reworked at some time to move the logic from the job into the main operator. So I don't want to add new fields to the CRD that would become obsolet.
As I stated in the issue, the snapshot config job can and should reuse resources from the initHelper, keeping in mind it's a temporary construct anyway.

**Comment by ramonheras** [comment]
Hi @swoehrl-mw, I implemented the feature this way, because it seemed the most consistent approach for me, but I get your point. I've updated the PR with the changes you requested. Let me know if something else is missing. Have a great day!!

**Comment by ramonheras** [comment]
@swoehrl-mw should I change something else?

**Comment by jplanckeel** [comment]
👋 Hello, i up this subject. It's cool for security and exploitation to have limit on job. are there any changes still to be made in the PR?

**Comment by ramonheras** [comment]
Hi @jplanckeel, This PR is missing some unit tests, but I stopped the development until it is clear what's going to happen with the repo after the acquisition of Opster by Elastic

**Comment by prudhvigodithi** [comment]
Adding @salyh @pchmielnik @swoehrl-mw @jochenkressin to please review this PR.
Thanks 
@bbarani

**Review by swoehrl-mw — CHANGES_REQUESTED** [review]
To quote myself:
> Please make sure all points from the [guideline](https://github.com/Opster/opensearch-k8s-operator/blob/main/docs/developing.md#submitting-a-pr) are present.
> 
>     * The new funcationality needs a unittest, even if it is just a simple one to check configured resources are part of the generated job/sts definition (there should be similar tests already).
> 
>     * The changes to the CRD are not copied into the operator helm chart
> 
>     * The change needs to be documented in the userguide

**Comment by bbarani** [comment]
> [ramonheras](/ramonheras)

@ramonheras, can you please work on these open items?

**Comment by ramonheras** [comment]
@bbarani, yess !! Now that I see that everything is up and running I'm willing finish this feature.

**Comment by ramonheras** [comment]
> (1) The new funcationality needs a unittest, even if it is just a simple one to check configured resources are part of the generated job/sts definition (there should be similar tests already).
> (2) The changes to the CRD are not copied into the operator helm chart
> (3) The change needs to be documented in the userguide

@swoehrl-mw. I totally agree with points 1 and 3, the need to have this feature available as soon as possible made me overlook that. But I don't understand point 2, could you expand on that?

File: charts/opensearch-operator/files/crd-servicemonitor.yaml
Comment (swoehrl-mw) [inline charts/opensearch-operator/files/crd-servicemonitor.yaml:1]: This is an external CRD that cannot be part of the chart

Comment (ramonheras) [reply charts/opensearch-operator/files/crd-servicemonitor.yaml:1]: Just doble checked and you're right, the CRD is not used in the Helm Chart, removing it. Thanks!

**Review by swoehrl-mw — APPROVED** [review]

**Pull request merged**